robotsim.py

```diff
@@ -452,15 +371,19 @@ def generate_map():
     if robot:
         myfont = pygame.font.SysFont('Arial', 12)
         textsurface = myfont.render('Movements = ' + str(robot.movements), False, (0, 0, 0))
-        gameDisplay.blit(textsurface,(display_width-pixel_constant*1.2,0))
+        gameDisplay.blit(textsurface,(pixel_constant*8 + pixel_constant*0.2, pixel_constant*0.2))
         textsurface = myfont.render('Points = ' + str(robot.points), False, (0, 0, 0))
-        gameDisplay.blit(textsurface,(display_width-pixel_constant*1.2,0.2*pixel_constant))
+        gameDisplay.blit(textsurface,(pixel_constant*8 + pixel_constant*0.2, 0.4*pixel_constant))
+        rosa_de_los_vientos = pygame.transform.scale(rosa_de_los_vientos, (int(pixel_constant*2), int(pixel_constant*2)))
+        gameDisplay.blit(rosa_de_los_vientos, (pixel_constant*8 + pixel_constant*0.2, pixel_constant))
     else:
         myfont = pygame.font.SysFont('Arial', 12)
         textsurface = myfont.render('Movements = 0', False, (0, 0, 0))
-        gameDisplay.blit(textsurface,(display_width-pixel_constant*1.2,0))
+        gameDisplay.blit(textsurface,(pixel_constant*8 + pixel_constant*0.2, pixel_constant*0.2))
         textsurface = myfont.render('Points = 0', False, (0, 0, 0))
-        gameDisplay.blit(textsurface,(display_width-pixel_constant*1.2,0.2*pixel_constant))
+        gameDisplay.blit(textsurface,(pixel_constant*8 + pixel_constant*0.2, 0.4*pixel_constant))
+        rosa_de_los_vientos = pygame.transform.scale(rosa_de_los_vientos, (int(pixel_constant*2), int(pixel_constant*2)))
+        gameDisplay.blit(rosa_de_los_vientos, (pixel_constant*8 + pixel_constant*0.2, pixel_constant))
```

Comment:
rosa_de_los_vientos does not depend on robot. It can be called just once outside the if robot.